modin/extension/__init__.py

```diff
@@ -0,0 +1,11 @@
+from .extensions import (
```

Comment:
Should we move `modin/extension` to `modin/pandas/api/extensions`?

Reply:
I am imagining eventually adding API extensions for `numpy` and even others.

Reply:
@sfc-gh-dpetersohn Users will most likely look for these functions in the same import path as in pandas. I think this is a good enough reason to make `from modin.pandas.api.extensions import register_*` work.

Reply:
If we wanted to add other API extentions, say for numpy, we would put them in `modin.numpy.api.extensions`. @sfc-gh-dpetersohn, what do you think?

Reply:
Sure, we can move it.